cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/async/GetOptimizationProposalsRunnable.java

```diff
@@ -37,9 +42,12 @@
       return _kafkaCruiseControl.getOptimizationProposals(_goals,
                                                           _modelCompletenessRequirements,
                                                           _future.operationProgress(),
-                                                          _allowCapacityEstimation);
+                                                          _allowCapacityEstimation,
+                                                          _excludedTopics);
     } else {
-      return _kafkaCruiseControl.getOptimizationProposals(_future.operationProgress(), _allowCapacityEstimation);
+      return _kafkaCruiseControl.getOptimizationProposals(_future.operationProgress(),
+                                                          _allowCapacityEstimation,
+                                                          _excludedTopics);
```

Comment:
Hey Efe, I do not think it is a good idea to use this branch if `excluded_topics` is set in request.

First, you never use the cached result if this [parameter is set](https://github.com/linkedin/cruise-control/blob/4c1489d8b0dd1e82870db28d55eabfdd69cf8775/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java#L327).so you have to do the expensive calculation anyway.

second, if `_numPrecomputingThreads` >0, you refer to [precomputing thread's result](https://github.com/linkedin/cruise-control/blob/4c1489d8b0dd1e82870db28d55eabfdd69cf8775/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java#L348), [which](https://github.com/linkedin/cruise-control/blob/4c1489d8b0dd1e82870db28d55eabfdd69cf8775/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java#L216) does not take `excluded_topics` into consideration, so you actually get a wrong result.